bitgouel-api/src/main/kotlin/team/msg/domain/lecture/service/LectureServiceImpl.kt

```diff
@@ -44,4 +48,48 @@ class LectureServiceImpl(
 
         lectureRepository.save(lecture)
     }
+
+    /**
+     * 강의 개설 신청을 수락하는 비지니스 로직입니다.
+     * @param UUID
+     */
+    override fun approveLecture(id: UUID) {
+        val lecture = queryLecture(id)
+
+        if(lecture.approveStatus == ApproveStatus.APPROVED)
+            throw AlreadyApprovedLectureException("이미 개설 신청이 승인된 강의입니다. info : [ uuid = $id ]")
```

Comment:
info에 uuid = id라고 표기하는 것 보단 lectureId = $id로 표기하는게 맞지 않을까 합니다.

```suggestion
        if(lecture.approveStatus == ApproveStatus.APPROVED)
            throw AlreadyApprovedLectureException("이미 개설 신청이 승인된 강의입니다. info : [ lectureId = $id ]")
```

Reply:
추가로 향후 혹은 지금 approve와 reject에서 검증하는 로직이 똑같은데 Spring AOP를 사용해서 reject, approve의 pointcut을(어노테이션이나 execution 등으로) 잡고 ApproveStatus 검증하는 Aspect Advice를 추가해봐도 괜찮을 것 같아요. 스프링 AOP, AspectJ에 대해 알아보셨으면 좋겠습니다.

Reply:
관련해서 공부 후에 빠르게 추가해보도록 하겠습니다!